pkgs/build-support/go/module.nix

```diff
@@ -307,12 +326,17 @@ let
 
     disallowedReferences = lib.optional (!allowGoReference) go;
 
-    passthru = { inherit go goModules vendorHash; } // passthru;
+    passthru = {
+      inherit go;
+      # Canonicallize `overrideModAttrs` as an attribute overlay.
+      # `passthru.overrideModAttrs` will be overridden
+      # when users want to override `goModules`.
+      overrideModAttrs = toExtension overrideModAttrs;
```

Comment:
Since this is the only place that `toExtension` is used, perhaps you'd like to put it inline here? It might make the diff a bit smaller..

Reply:
Quoting myself: 

> Since this is the only place that `toExtension` is used, perhaps you'd like to put it inline here? It might make the diff a bit smaller..

What do you think about this idea @ShamrockLee ?

Reply:
If this function works well, I plan to move it into `lib.fixedPoints`, so I defined it separately.

Reply:
> If this function works well, I plan to move it into `lib.fixedPoints`, so I defined it separately.

OK sounds like a good idea to me.